packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -240,6 +240,29 @@ export default declare((api, opts) => {
     }
   }
 
+  function hasMoreThanOneBinding(node) {
```

Comment:
Since this function is independent and complex, we could extract it to a new module and create unit tests for it.

The input can be created from source => (babel parser) => ast, the test function is `hasMoreThanOneBinding`. It will be easier to reason about whether `hasMoreThanOneBinding` works as expected than going through the messy fixture outputs.

See https://github.com/babel/babel/blob/main/packages/babel-types/test/retrievers.js for an example test setup.

Reply:
I have something like this working.

would you be amenable to something like this?

```js
import { parse } from "@babel/parser";
import hasMoreThanOneBinding from "../lib/hasMoreThanOneBinding";

function getBody(program) {
  return parse(program, { sourceType: "module" }).program.body;
}
describe("hasMoreThanOneBinding", function () {
  it.each([
    [
      "variable declarations",
      getBody("const { x15: [...{ ...y15 }] } = z();"),
      false,
    ],
  ])("%s", (_, ast, expectedResult) => {
    const result = hasMoreThanOneBinding(ast);
    expect(result).toEqual(expectedResult);
  });
});

```

Happy to do this. I would like to get this fix in sooner as we've had a few regressions due to this bug at work.

Reply:
That looks great! You may also put `getBody` in the test functions so the test cases can be
```
[ "const { x15: [...{ ...y15 }] } = z();", false ],
[ "const { x15: [...{ ...y15, z15 }] } = z();", true ],
...
```

Reply:
@JLHwung made a distinct test file with these test cases. I copied the test cases from input.js and added the expected boolean return.